python/src/netlist.py

```diff
@@ -41,6 +41,16 @@ def contents(self) -> str:
             self._cached = self.file.read()
         return self._cached
 
+    def json_repr(self):
+        return {"__BaseNetlistFile__": True, **self.__dict__}
```

Comment:
nit: I'd prefer `_cached` not be written to json

Reply:
Nevermind i see its ignore in `from_json` 👍

Reply:
Yeah I was going back and forth about whether to dump it, I think I might remove it again since now  that I think of it, it keeps a pointer to the netlist it is based on